File: util/android/src/main/java/com/dmi/util/opengl/GLRenderer.java
package com.dmi.util.opengl;

public interface GLRenderer {
    void onSurfaceCreated();

    void onSurfaceChanged(int width, int height);

    /**
     * Free resources, created in onSurfaceCreated. May be invoke multiple times after onSurfaceCreated.
     */
    void onFreeResources();

    void onDrawFrame();
}

File: util/android/src/main/kotlin/com/dmi/util/opengl/GLRenderer.kt
package com.dmi.util.opengl

interface GLRenderer {
    fun onSurfaceCreated()

    fun onSurfaceChanged(width: Int, height: Int)

    /**
     * Free resources, created in onSurfaceCreated. May be invoke multiple times after onSurfaceCreated.
     */
    fun onFreeResources()

    fun onDrawFrame()
}
